samcli/local/apigw/local_apigw_service.py

```diff
@@ -319,8 +324,20 @@ def _request_handler(self, **kwargs):
                     route_key,
                 )
             else:
+
+                # The OperationName is only sent to the Lambda Function from API Gateway V1(Rest API).
+                # For Http Apis (v2), API Gateway never sends the OperationName.
+                if route.event_type == Route.API:
+                    operation_name = route.operation_name
+                else:
+                    operation_name = None
                 event = self._construct_v_1_0_event(
-                    request, self.port, self.api.binary_media_types, self.api.stage_name, self.api.stage_variables
+                    request,
+                    self.port,
+                    self.api.binary_media_types,
+                    self.api.stage_name,
+                    self.api.stage_variables,
+                    operation_name,
```

Comment:
We can change this later since I don't think we have push access to remote. Keep nesting like this are very challenging to maintain. Instead we should be doing "if some condition: do a thing", "if other thing: do this thing". 

So a much more maintainable way is:

```
# in all other cases but the if's below
create event

if event_type is `Route.API`: 
  create event

if event_type is Route.HTTP:
   create event

if event_type is Route.Http and some condition:
  create event
```

Its much easier to read, as you expand you don't have a magical path that defaults to something (sometimes that is better but can lead to unexpected bugs).

Reply:
Thanks for the suggestion. Updated.